File: 19-RemoveNthNodeFromEndOfList/19-RemoveNthNodeFromEndOfList.java
// Last updated: 02/01/2026, 10:11:04
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode temp=new ListNode();
        int len=0;
        temp=head;
        while(temp!=null){
            temp=temp.next;
            len++;
        }
        
        if(len==0){
            return null;
        }
        if(n==len){
            return head.next;
        }
        ListNode hold=new ListNode();
        temp=head;
        hold=head;
        for(int i=0;i<(len-n);i++){
            hold=temp;
            temp=temp.next;
        }
        hold.next=temp.next;
        return head;
    }
}

File: 19-RemoveNthNodeFromEndOfList/19-RemoveNthNodeFromEndOfList.c
// Last updated: 02/01/2026, 10:11:11
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* removeNthFromEnd(struct ListNode* head, int n) {
    struct ListNode* temp=(struct ListNode*)malloc(sizeof(struct ListNode));
    temp=head;
    int length=0;
    while(temp!=0){
        temp=temp->next;
        length++;
    }
    printf("%d",length);
    int i=1;
    struct ListNode* hold=(struct ListNode*)malloc(sizeof(struct ListNode));
    temp=head;  
    hold=head;
    if(length==0){
        free(head);
        return NULL;
    }
    if(n==length){
        hold=head->next;
        free(head);
        return hold;
    }
    for(int i=0;i<(length-n);i++){
        hold=temp;
        temp=temp->next;
    }
    hold->next=temp->next;
    return head;
}